lib/ansible/modules/clustering/kubevirt/kubevirt_template.py

```diff
@@ -0,0 +1,374 @@
+               caches if those are exposed by the underlying volume implementation."
+        type: dict
+    default_nic:
+        description:
+            - "The goal of default network is similar to I(default_disk) and should be used as a template
+               to ensure OS compatibility and performance."
+        type: dict
+    default_network:
+        description:
+            - "The goal of default network is similar to I(default_volume) and should be used as a template
+               that specifies performance and connection parameters (L2 bridge for example)"
+        type: dict
+    icon_class:
+        description:
+            - "An icon to be displayed with your template in the web console. Choose from our existing logo
+            icons when possible. You can also use icons from FontAwesome. Alternatively, provide icons through
+            CSS customizations that can be added to an OpenShift Container Platform cluster that uses your template.
+            You must specify an icon class that exists, or it will prevent falling back to the generic icon."
+        type: str
+    parameters:
+        description:
+            - "Parameters allow a value to be supplied by the user or generated when the template is instantiated.
+            Then, that value is substituted wherever the parameter is referenced. References can be defined in any
+            field in the objects list field. This is useful for generating random passwords or allowing the user to
+            supply a host name or other user-specific value that is required to customize the template."
+            - "More information can be foud at: U(https://docs.openshift.com/container-platform/3.6/dev_guide/templates.html#writing-parameters)"
+        type: list
+    version:
+        description: ""
+            - "Specify the version of the template."
+        type: str
+
+extends_documentation_fragment:
+  - k8s_auth_options
+  - k8s_resource_options
+
+requirements:
+  - python >= 2.7
+  - openshift >= 0.8.2
+'''
+
+EXAMPLES = '''
+- name: Create template 'mytemplate'
+  kubevirt_template:
+    state: present
+    name: myvmtemplate
+    display_name: Generic cirros template
+    description: Basic cirros template
+    long_description: Verbose description of cirros template
+    provider_display_name: Just Be Cool, Inc.
+    documentation_url: http://theverycoolcompany.com
+    support_url: http://support.theverycoolcompany.com
+    icon_class: icon-linux
+    namespace: templates
+    default_disk:
+      - name: default-disk
+        disk:
+          bus: virtio
+    default_nic:
+      - name: default-interface
+        model: virtio
+    default_network:
+      - name: default-network
+        resource:
+          resourceName: bridge.network.kubevirt.io/cnvmgmt
+    default_volume:
+      - name: default-volume
+        containerDisk:
+          image: kubevirt/cirros-container-disk-demo:latest
+    objects:
+      - name: ${NAME}
```

Comment:
This is not a valid VM object in kubevirt form so the description above is inaccurate. You should somehow describe the valid simplified structures (by pointing to the other modules or something)

Reply:
Thanks, I've added documentation, where user should look for structure.